modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/service/test/JournalFolderLocalServiceTest.java

```diff
@@ -50,6 +55,77 @@ public class JournalFolderLocalServiceTest {
 	@Before
 	public void setUp() throws Exception {
 		_group = GroupTestUtil.addGroup();
+		_user = TestPropsValues.getUser();
+	}
+
+	@Test
+	public void testAddJournalFolderWithExternalReferenceCode()
+		throws Exception {
+
+		String externalReferenceCode = RandomTestUtil.randomString();
+
+		JournalFolder folder = addJournalFolder(
+			externalReferenceCode, RandomTestUtil.randomString(),
+			RandomTestUtil.randomString());
+
+		Assert.assertEquals(
+			externalReferenceCode, folder.getExternalReferenceCode());
+	}
+
+	@Test
+	public void testAddJournalFolderWithoutExternalReferenceCode()
+		throws Exception {
+
+		JournalFolder folder = addJournalFolder(
+			null, RandomTestUtil.randomString(), RandomTestUtil.randomString());
+
+		Assert.assertEquals(
+			folder.getExternalReferenceCode(),
+			String.valueOf(folder.getFolderId()));
+	}
+
+	@Test
+	public void testDeleteJournalFolderByExternalReferenceCode()
```

Comment:
As we only added logic to create and receive journal folders by external reference code, I would not add this method.

Reply:
+1